packages/aws-cdk-lib/aws-codepipeline-actions/lib/codecommit/source-action.ts

```diff
@@ -169,15 +207,21 @@ export class CodeCommitSourceAction extends Action {
 
     const createEvent = this.props.trigger === undefined ||
       this.props.trigger === CodeCommitTrigger.EVENTS;
-    if (createEvent) {
-      const eventId = this.generateEventId(stage);
+    const eventId = this.generateEventId(stage);
+
+    if (createEvent && this.props.customEventRule === undefined) {
```

Comment:
If both `createEvent` and `this.props.customEventRule` are defined then neither of these will be run. It looks like `createEvent` is set if `trigger` is set to `CodeCommitTrigger.EVENTS` so you'll have to somehow enforce that `trigger` is not be set when a `customEventRule` is passed in and maybe reflect that somewhere in the docs.

Reply:
@paulhcsun I think it does trigger the `else if`- statement in line 220 since `customEventRule` will be true. Otherwise, if both set to `undefined`, the trigger becomes `CodeCommitTrigger.POLL` (line 253)